### PR TITLE
Jason/fix reflow issue

### PR DESCRIPTION
**Pull request by duidae**
白頁初步判斷是某些行為引發forced synchronous layouts(layout thrashing or reflow) [ref](https://gist.github.com/paulirish/5d52fb081b3570c81e3a)，使效能較差的平台(i.e. mobile + safari)在文章頁會明顯看見比較長時間的空白:

1. [已刪除]progress bar(scroll顯示進度條): scroll時呼叫getBoundingClientRect(), [ref](https://stackoverflow.com/questions/13234214/reflow-layout-performance-for-large-application)

2. mobile aside(向下scroll顯示bookmark): scroll時讀取window.scrollY
Screen Shot 2022-05-26 at 11 43 12 PM

3. header(向上/下scroll顯示/隱藏header): scroll時讀取window.pageYOffset
截圖 2022-05-25 下午11 11 57

4. 還有一種行為會引發layout, 但觸發點還未找到
Screen Shot 2022-05-27 at 10 56 37 PM

2, 3之解法: 用requestAnimationFrame() 
https://www.html5rocks.com/en/tutorials/speed/animations/
https://developer.mozilla.org/en-US/docs/web/api/document/scroll_event#Example
https://codertw.com/%E7%A8%8B%E5%BC%8F%E8%AA%9E%E8%A8%80/692855/#outline__3_2_1
https://yonatankra.com/layout-reflow/

### CODE REVIEW

File: packages/universal-header/src/containers/header.js [hunk 1]
```diff
@@ -92,18 +92,33 @@ class Container extends React.PureComponent {
 
   componentWillUnmount() {
     window.removeEventListener('scroll', this.handleScroll)
+    this.lastKnownPageYOffset = 0
+    this.ticking = false
```
Comment (taylrj) [inline packages/universal-header/src/containers/header.js:96]: How about assign `null` to them in `componentWillUnmount` to help garbage collection ?

File: packages/react-article-components/src/components/article-page.js [hunk 2]
```diff
@@ -326,21 +324,23 @@ export default class Article extends PureComponent {
...
   }
 
   componentWillUnmount() {
     window.removeEventListener('scroll', this.onScroll)
-
+    this.lastKnownScrollPosition = 0
+    this.ticking = false
     this.scrollPosition = {
       y: 0,
     }
```
Comment (taylrj) [inline packages/react-article-components/src/components/article-page.js:346]: How about assigning them to null in componentWillUnmount to help garbage collection?

**Comment by taylrj** [comment]
感謝發現 reflow 的原因，太強大了！！！  🙇🏻

我這裡還有看到幾個，但這應該不是我們現階段能改的了：

image

image

image

**Comment by taylrj** [comment]
我在 iOS 15.4.1 Safari 上實測這一包還是有白畫面（見下方螢幕錄影），不過我覺得這也是個改善 performance 的 refactor，可以先上 staging 再測測看。


https://user-images.githubusercontent.com/6375655/170910522-b1acdc09-9cf8-4cda-a031-438d46e79423.MP4

File: packages/universal-header/src/containers/header.js (hunk 1 above)
Comment (duidae) [reply packages/universal-header/src/containers/header.js]: fixed. btw, just curious is it necessary to clean those member variables in `componentWillUnmount`, I just copy the style without much thinking XD

File: packages/react-article-components/src/components/article-page.js (hunk 2 above)
Comment (duidae) [reply packages/react-article-components/src/components/article-page.js:346]: fixed.

File: packages/universal-header/src/containers/header.js (hunk 1 above)
Comment (taylrj) [reply packages/universal-header/src/containers/header.js]: In my view, yes. But it is a convention to me, more or less. 

So if there are some refs which disagree with such clean, we could have further discussion about this issue.

**Comment by taylrj** [comment]
LGTM

**Pull request merged**